hazelcast/src/main/java/com/hazelcast/client/impl/ClientEndpointImpl.java

```diff
@@ -234,6 +234,11 @@ public void clearAllListeners() {
         removeListenerActions.clear();
     }
 
+    @Override
+    public boolean resourcesExist() {
+        return removeListenerActions.isEmpty() && transactionContextMap.isEmpty();
```

Comment:
Is the name of the method correct or is there an issue with the logic? I would expect the following logic based on the method name.
```
return !removeListenerActions.isEmpty() || !transactionContextMap.isEmpty();
```